include/alpaka/core/Decay.hpp

```diff
@@ -29,3 +29,10 @@
 #else
 #    define ALPAKA_DECAY_T(Type) std::decay_t<Type>
 #endif
+
+namespace alpaka
+{
+    //! Provides a decaying wrapper around std::is_same. Example: is_decayed_v<volatile float, float> returns true.
+    template<typename T, typename U>
+    inline constexpr auto is_decayed_v = std::is_same_v<ALPAKA_DECAY_T(T), U> || std::is_same_v<T, ALPAKA_DECAY_T(U)>;
```

Comment:
It appears that clang 5 and 6 are not treating `inline` variables correctly here.